src/bundle.c

```diff
@@ -135,26 +99,22 @@ bool is_installed_bundle(const char *bundle_name)
 	return ret;
 }
 
-/* When loading all tracked bundles into memory, they happen
- * to be hold in subs global var, for some reasons it is
- * needed to just pop out one or more of this loaded tracked
- * bundles, this function search for bundle_name into subs
- * struct and if it found then free it from the list.
- */
-static enum swupd_code unload_tracked_bundle(const char *bundle_name, struct list **subs)
+static enum swupd_code unload_manifest(const char *bundle_name, struct list **bundles_installed, struct list **bundles_to_remove)
```

Comment:
This a generic list operation too. You are removing one searching one element from one list, removing it and moving to another one. We could have a list_remove(list, comparison_fn) and then:

    item = list_remove(bundles_installed, manifest_cmp_fn);
    bundles_to_remove = list_prepend_data(bundles_to_remove, item);

Reply:
good idea!

Reply:
done